### DISCUSSION TITLE
:zany_face: prisma/.devcontainer can't connect to databases... why?

### DISCUSSION THREAD

**Discussion by bandinopla** [prisma/prisma#14826] [Q&A] [answered]
My goal is to run the jest tests of a package...

i've opened the prisma repo inside a Docker Container and did all the setup, but when i try to run a test that needs access to a database, it says it fails to reach the database. For example, this:

```
/workspaces/prisma/packages/migrate > $ pnpm run test -- MigrateDev.test
```

produces...

```
Rejected to value: [Error: P1001: Can't reach database server at `localhost`:`3306`·
Please make sure your database server is running at `localhost`:`3306`.]
```

from within the container i can't launch the `docker-compose.yml` in the **docker** folder. So i opened the project on my windows enviroment and run it form there. But one thing i noticed is that the name of the images change (?) in the yaml for example the mysql service is called "mysql" but in my docker desktop it is renamed to "docker_mysql_1" mabe that's the issue?

Also running `printenv` from within the peroject container doesn't show any of the enviroment variables realting to database connection strings...

**How does one setup a dev enviroment to work on prisma files and run the jest tests?**

**Comment by bandinopla** [comment]
something that i find confusing about the **.devcontainer** folder is that if you clone and openm the repo on a virtual container you can't run tests because the docker compose file in the **docker** folder can't be run and if i run it on my machine, the services it publish are not accessible from within the opened container...

and here https://github.com/prisma/prisma/blob/main/.buildkite/test/docker-compose.14.yml you can see that indeed that is the correct way to launch a container "connected" with the database services...

so what was is point of the **.devcontainer** setup if it can't access the test databases??

**Reply by janpio** [reply]
What is the `.devcontainer` folder?

**Reply by bandinopla** [reply]
this is the devcontainer file: https://github.com/prisma/prisma/blob/main/.devcontainer/devcontainer.json Visual Studio Remote plugin picks that up and opens the entire project inside of a container. That works fine.

Before that, i went to my local copy of prisma's source folder and manually `docker-compose up` this https://github.com/prisma/prisma/blob/main/docker/docker-compose.yml again that works fine, i can see all the database services running on my machine...

**BUT** when i try to run a test, inside the Visual Studio Contained project, i get the error "can't connect to XXXX" 

my guess is that the container doesn't know how to access the services running on my local pc... but it is weird i had to change something since im working with a fresh clone of the prisma repo... how do the developers work on the diferent packages if they need access to those database services??

**Comment by bandinopla** [comment, marked as answer]
found the solution here: https://stackoverflow.com/questions/73379167/how-to-connect-a-dev-container-to-another-container/73383935#73383935

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
